optimizely/config_manager.py

```diff
@@ -0,0 +1,198 @@
+# Copyright 2019, Optimizely
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# 
+# http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import http
+import requests
+import threading
+import time
+from requests import exceptions as requests_exceptions
+
+from optimizely import exceptions
+from optimizely import logger as _logging
+from optimizely import project_config
+from optimizely.error_handler import NoOpErrorHandler as noop_error_handler
+from optimizely.helpers import enums
+
+
+class BaseConfigManager(object):
+  """ Base class for Optimizely's config manager. """
+
+  def get_config(self):
+    """ Get config for use by optimizely.Optimizely.
+    The config should be an instance of project_config.ProjectConfig."""
+    raise NotImplementedError
+
+
+class StaticConfigManager(BaseConfigManager):
+  """ Config manager that returns ProjectConfig based on provided datafile. """
+
+  def __init__(self,
+               datafile,
+               logger=None,
+               error_handler=None):
+    self.logger = logger or _logging.adapt_logger(logger or _logging.NoOpLogger())
```

Comment:
Can we add a docstring for what these arguments are?

Reply:
Will add.